docs/source/distributed.rst

```diff
@@ -293,6 +293,33 @@ check whether the process group has already been initialized use :func:`torch.di
 
 .. autofunction:: get_world_size
 
+Shutdown
+--------
+
+It is important to clean up resources on exit by calling :func:`destroy_process_group`.
+
+The simplest pattern to follow is to destroy every process group and backend by calling
+:func:`destroy_process_group()` with the default value of None for the `group` argument, at a
+point in the training script where communications are no longer needed, usually near the
+end of main().  The call should be made once per trainer-process, not at the outer
+process-launcher level.
+
+if :func:`destroy_process_group` is not called, especially when there are multiple process-groups
+in the application e.g. for N-D parallelism, hangs on exit are possible.  This is because the
+destructor for ProcessGroupNCCL calls ncclCommAbort, which must be called collectively, but
```

Comment:
"destructor for ProcessGroupNCCL calls ncclCommAbort" -> "destructor for ProcessGroupNCCL would call ncclCommAbort if this NCCL API has NOT been called by destroy_process_group before destruction"